### PR TITLE
DW-535: Added API request handler

### PR DESCRIPTION
**Pull request by rimi-itk**
https://jira.itkdev.dk/browse/DW-535

Adds a simple webform handler that sends webform id and submission uuid to an API endpoint.

### CODE REVIEW

File: web/modules/custom/os2forms_api_request_handler/src/Plugin/AdvancedQueue/JobType/PostSubmission.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Drupal\os2forms_api_request_handler\Plugin\AdvancedQueue\JobType;
+
+use Drupal\advancedqueue\Job;
+use Drupal\advancedqueue\JobResult;
+use Drupal\advancedqueue\Plugin\AdvancedQueue\JobType\JobTypeBase;
+
```
Comment (jekuaitk) [inline web/modules/custom/os2forms_api_request_handler/src/Plugin/AdvancedQueue/JobType/PostSubmission.php:8]: Remove this.

File: web/modules/custom/os2forms_api_request_handler/src/Plugin/AdvancedQueue/JobType/PostSubmission.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\os2forms_api_request_handler\PostHelper;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Archive document job.
```
Comment (jekuaitk) [inline web/modules/custom/os2forms_api_request_handler/src/Plugin/AdvancedQueue/JobType/PostSubmission.php:14]: Needs an update.

File: web/modules/custom/os2forms_api_request_handler/src/Plugin/AdvancedQueue/JobType/PostSubmission.php [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get(PostHelper::class)
+    );
+  }
+
+  /**
+   * Processes the ArchiveDocument job.
```
Comment (jekuaitk) [inline web/modules/custom/os2forms_api_request_handler/src/Plugin/AdvancedQueue/JobType/PostSubmission.php:55]: Needs an update.

File: web/modules/custom/os2forms_api_request_handler/src/PostHelper.php [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+  /**
+   * The client.
+   *
+   * @var \GuzzleHttp\Client
+   */
+  private Client $client;
+
+  /**
+   * Constructs an ArchiveHelper object.
```
Comment (jekuaitk) [inline web/modules/custom/os2forms_api_request_handler/src/PostHelper.php:41]: Needs an update.

File: web/modules/custom/os2forms_api_request_handler/src/PostHelper.php [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+      $submission = $this->getSubmission($submissionId);
+      if (NULL === $submission) {
+        throw new ApiHandlerException(sprintf('Cannot load submission %s', $submissionId));
+      }
+
+      $webform = $submission->getWebform();
+
+      $links = [
+        'sender' => $this->generateUrl('<front>'),
```
Comment (jekuaitk) [inline web/modules/custom/os2forms_api_request_handler/src/PostHelper.php:65]: `<front>` looks like a funny route name, is it perhaps some shortcut or something?

**Review by jekuaitk — CHANGES_REQUESTED** [review]

File: web/modules/custom/os2forms_api_request_handler/src/PostHelper.php (hunk 5 above)
Comment (rimi-itk) [reply web/modules/custom/os2forms_api_request_handler/src/PostHelper.php:65]: https://github.com/drupal/drupal/blob/9.5.x/core/modules/system/system.routing.yml#L394-L399

**Review by jekuaitk — APPROVED** [review]

**Review by martinyde — APPROVED** [review]

**Review by jekuaitk — APPROVED** [review]

**Pull request merged**